frame.test.js

```diff
@@ -0,0 +1,160 @@
+const Frame = require('./frame')
+const Scorecard = require('./scorecard')
+
+describe('Frame', () => {
+  let frame
+  let scorecard
+
+  beforeEach(() => {
+    frame = new Frame()
+    scorecard = new Scorecard()
+  })
+
+  describe('a simple frame of two rolls with no strikes or spares', () => {
+    it('gives the score at the start of a frame', () => {
+      expect(frame.calculateTotalScore()).toBe(0)
+    })
+
+    it('captures score of first roll', () => {
+      frame.enterFirstRollScore(3, scorecard)
+      expect(frame.firstRollScore).toEqual(3)
+    })
+
+    it('adds first roll score to frame total', () => {
+      frame.enterFirstRollScore(3, scorecard)
+      expect(frame.calculateTotalScore()).toEqual(3)
+    })
+
+    it('captures score of second roll', () => {
+      frame.enterSecondRollScore(2, scorecard)
+      expect(frame.secondRollScore).toEqual(2)
+    })
+
+    // This is taken from the example of the first two rolls on example_ten_pin_scoring.png
+    it('knows a roll of one on the first roll and four on the second roll scores a total of five', () => {
```

Comment:
This also gets indirectly tested on line 53 when calculating the score for a spare, which then makes this test obsolete. Take DRY into consideration.